pennylane/gradients/parameter_shift.py

```diff
@@ -1264,7 +1499,22 @@ def processing_fn(results):
 
         return gradient_tapes, processing_fn
 
-    return gradient_tapes, fn
+    def proc_with_validation(results):
+        """Assume if a ValueError is raised during the computation, then
+        shot vectors are used and the shots argument was not set correctly."""
+        try:
+            res = fn(results)
+        except (ValueError, TypeError) as e:

```

Comment:
Is it temporary to make sure people pass shots to the gradient transform?

Reply:
Well, temporary as in until the `shots` argument is required. Once we have `shots` info on the tape, it could be removed.

> make sure people pass shots to the gradient transform

Yes, this would be the goal - that users don't see cryptic error messages just because they haven't passed in shots.